R/dup.R

```diff
@@ -5,14 +5,23 @@
 #'
 #' @inheritParams str_detect
 #' @param times Number of times to duplicate each string.
+#' @param sep String to insert between each duplicate.
 #' @return A character vector the same length as `string`/`times`.
 #' @export
 #' @examples
 #' fruit <- c("apple", "pear", "banana")
 #' str_dup(fruit, 2)
 #' str_dup(fruit, 1:3)
 #' str_c("ba", str_dup("na", 0:5))
-str_dup <- function(string, times) {
-  vctrs::vec_size_common(string = string, times = times)
-  stri_dup(string, times)
+str_dup <- function(string, times, sep = NULL) {
+  input <- vctrs::vec_recycle_common(string = string, times = times)
+  check_string(sep, allow_null = TRUE)
```

Comment:
I think it makes sense to move the other input check up here so that all the input checks are together.